src/date-adapters/luxon/index.ts

```diff
@@ -0,0 +1,187 @@
+    );
+  }
+
+  function differenceInMinutes(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('minutes')
+    );
+  }
+
+  function differenceInSeconds(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('seconds')
+    );
+  }
+
+  function endOfDay(date: Date | number): Date {
+    return coerceDateTime(date).endOf('day').toJSDate();
+  }
+
+  function endOfMonth(date: Date | number): Date {
+    return coerceDateTime(date).endOf('month').toJSDate();
+  }
+
+  function endOfWeek(
+    date: Date | number,
+    options?: {
+      weekStartsOn?: number;
+    }
+  ): Date {
+    return coerceDateTime(date)
+      .endOf('week', { useLocaleWeeks: true })
+      .plus({ days: /* istanbul ignore next */ options?.weekStartsOn ?? 0 })
+      .toJSDate();
+  }
+
+  function getDay(date: Date | number): number {
+    return coerceDateTime(date).weekday % 7;
+  }
+
+  /* istanbul ignore next */
+  function getMonth(date: Date | number): number {
+    return coerceDateTime(date).month % 12;
+  }
+
+  function isSameDay(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): boolean {
+    return coerceDateTime(dateLeft).hasSame(coerceDateTime(dateRight), 'day');
+  }
+
+  function isSameMonth(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): boolean {
+    return coerceDateTime(dateLeft).hasSame(coerceDateTime(dateRight), 'month');
+  }
+
+  function isSameSecond(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): boolean {
+    return coerceDateTime(dateLeft).hasSame(
+      coerceDateTime(dateRight),
+      'second'
+    );
+  }
+
+  function max(dates: (Date | number)[]): Date {
+    return luxon.DateTime.max(
+      ...dates.map((date) => coerceDateTime(date))
+    ).toJSDate();
+  }
+
+  function setHours(date: Date | number, hours: number): Date {
+    return coerceDateTime(date).set({ hour: hours }).toJSDate();
+  }
+
+  function setMinutes(date: Date | number, minutes: number): Date {
+    return coerceDateTime(date).set({ minute: minutes }).toJSDate();
+  }
+
+  function startOfDay(date: Date | number): Date {
+    return coerceDateTime(date).startOf('day').toJSDate();
+  }
+
+  function startOfMinute(date: Date | number): Date {
+    return coerceDateTime(date).startOf('minute').toJSDate();
+  }
+
+  function startOfMonth(date: Date | number): Date {
+    return coerceDateTime(date).startOf('month').toJSDate();
+  }
+
+  function startOfWeek(
+    date: Date | number,
+    options?: {
+      weekStartsOn?: number;
+    }
+  ): Date {
+    return coerceDateTime(date)
+      .startOf('week', {
+        useLocaleWeeks: true,
+      })
+      .plus({ days: /* istanbul ignore next */ options?.weekStartsOn ?? 0 })
```

Comment:
Should this use `.set({localWeekNumber: options?.weekStartsOn ?? 0})`?

Reply:
See endOfWeek. I would proceed in the same way.